File: level3/queue-to-do/Solution.java
class Solution {
    public static int solution(int start, int length) {
        int ans = start;
        
        for (int n=1; n<=length; n++) {
            int items = -n + length + 1;
            int first = (n-1) * length + start;
            for (int x=first; x<items+first; x++) {
                ans = ans^x;
            }
        }
        
        return ans^start;
    }

    public static void main(String[] args) {
      System.out.println(Solution.solution(Integer.valueOf(args[0]), Integer.valueOf(args[1])));
    }
}

File: level3/queue-to-do/solution.py
import sys

def solution(start, length):
    ans = start

    for n in range(1, length+1):
        items = -n + length + 1
        first = (n-1) * length + start
        for x in range(first, items+first):
            ans = ans^x

    return ans^start

# print(sys.argv)
print(solution(int(sys.argv[1]), int(sys.argv[2])))
